vaadin-renderer-flow-parent/vaadin-renderer-flow/src/main/java/com/vaadin/flow/data/renderer/ComponentDataGenerator.java

```diff
@@ -83,6 +83,9 @@ public void generateData(T item, JsonObject jsonObject) {
             nodeId = oldRenderedComponent.getElement().getNode().getId();
         } else {
             Component renderedComponent = createComponent(item);
+            if (renderedComponent == null) {
+                return;
```

Comment:
With virtual scrolling components, that re-use the same container-element, the old content doesn't get removed:

```java
grid.addComponentColumn(item -> {
    if (item.getFirstName().equals("Person 40")) {
        return null;
    }

    return new TextField("", item.getFirstName(), "");

}).setHeader("Test");
```

https://user-images.githubusercontent.com/1222264/207805170-0a6fec81-f481-40ba-a8da-3ec6cb005e38.mp4

Reply:
This is now something new to master 🤔

Reply:
ComponentRenderer was refactored for V24 to use `Lit` instead of a `<template>` internally. Could be that this part worked differently on V23.

Reply:
Do you have good ideas how to test this in the module itself?

Reply:
It depends on the fix whether this can be tested with a server-side test case or if it requires an IT. That may be tricky since there isn't an IT setup for ComponentRenderer in the module yet.

Reply:
Damn, I should have done this in the summer 🤯

Reply:
A (probably bit non-optimal) workaround could be to drop in a dummy div component in ComponentDataGenerator if the supplier returns null 🤷‍♂️

Reply:
Like this: #4366 

But I have no idea to test that in a reasonable way...

Reply:
Now using this dummyfix. The good thing (even if this uses slightly more bandwidth) is that it might be easier to backport (didn't check, but just a guess).